libraries/core_libs/consensus/src/pbft/block_proposer.cpp

```diff
@@ -32,9 +32,10 @@ bool SortitionPropose::propose() {
   }
 
   const auto proposal_period = dag_blk_mgr_->getProposalPeriod(propose_level);
+  const auto period_block_hash = db_->getPeriodBlockHash(proposal_period.first);
```

Comment:
This is wrong. proposal period is not finalized period. 
If proposal period is behind of finalized period, that's ok.
But if proposal period is ahead of finalized period, the block hash is not exist

Reply:
How could be proposal period ahead of finalized period?

Reply:
You right. From the current design, proposal period should always behind of finalized period